Scripts/RemediationScripts/Remediate-AddDiagnosticSettingsForLogicApps.ps1

```diff
@@ -0,0 +1,507 @@
+                $storageAccountName = "diagn"
+                # Storage Account name will be a concatenation of "diag" + name of the RG(first 5 characters) + Location(first 15 characters).
+                # Only the first 5 characters of the Resource Group name and first 15 characters of location will be considered, as there is a size limit of 24 characters for the name of a Storage Account.
+                # This will strip out non-alphanumeric characters as Storage Account names can only contain alphanumeric characters.
+
+                $storageAccountLocationSuffix = $location -replace "\W"
+
+                # This check is required, else, String::Substring() will throw an error for strings less than 15 characters.
+
+                if ($storageAccountLocationSuffix.Length -gt 15) {
+                    $storageAccountLocationSuffix = $storageAccountLocationSuffix.Substring(0, 15)
+                }
+
+                 $storageAccountRGNameSuffix = $resourceGroup -replace "\W"
+
+                # This check is required, else, String::Substring() will throw an error for strings less than 5 characters.
+
+                if ($resourceGroup.Length -gt 5) {
+                    $storageAccountRGNameSuffix = $storageAccountRGNameSuffix.Substring(0, 5)
+                }
+                $storageAccountName = -join ("diag",$storageAccountRGNameSuffix.ToLower(), $storageAccountLocationSuffix.ToLower() )
+
+
+                $storageAccount= $null
+
+                $storageAccount = Create-StorageAccountIfNotExists $storageAccountResourceGroupName $storageAccountName $location
+
+                if (($storageAccount | Measure-Object).Count -eq 0) 
+                {
+                    Write-Host "Error creating a Storage Account to store the diagnostic logs." -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Please ensure that you have sufficient permissions to create a Storage Account in this Resource Group." -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+                    break
+                }
+
+                #Selecting only 1 account as observed duplicate entries from PS command intermittently
+                $storageAccount = $storageAccount | Select-Object -First 1
+
+                Write-Host "Adding diagnostic settings for Logic App $($_.ResourceId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+                $log = @()
+                $log += New-AzDiagnosticSettingLogSettingsObject -Enabled $true -Category "WorkflowRuntime"
+                $temp= New-AzDiagnosticSetting -Name 'LogicApp-DiagnosticSetting' -ResourceId $_.ResourceId  -StorageAccountId $storageAccount.Id  -Log $log
+
+                    
+                $isDiagnosticSettingAdded = $false
+
+                $settings = Get-AzDiagnosticSetting -ResourceId $_.ResourceId
```

Comment:
Do we really need to make one more call, retuned object command should have same info?

Reply:
used retuned object from comamnd, thanks